misc/bazel/c_deps/BUILD.lz4.bazel

```diff
@@ -13,10 +13,25 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-"""A module initialising the third party dependencies for building OpenSSL"""
+"""Builds lz4."""
 
-load("@rules_perl//perl:deps.bzl", "perl_register_toolchains", "perl_rules_dependencies")
-
-def openssl_setup():
-    perl_rules_dependencies()
-    perl_register_toolchains()
+cc_library(
+    name = "lz4",
+    srcs = [
+        "lib/lz4.c",
+        "lib/lz4frame.c",
+        "lib/lz4hc.c",
+        "lib/xxhash.c",
+    ],
+    hdrs = [
+        "lib/lz4.c",
+        "lib/lz4.h",
+        "lib/lz4frame.h",
+        "lib/lz4hc.h",
+        "lib/xxhash.h",
+    ],
+    include_prefix = "lz4",
+    # Always compile with optimizations enabled, otherwise things are much too slow.
+    copts = ["-O3"],
```

Comment:
Stuff like this scares me a bit because it feels like we could be unaware or forget to set it for other C libs and then needlessly reduce the speed of some our our C execution. I think my ask for the future is that whatever lint we have that ensures the build.rs and bazel files are in sync covers these too.

Reply:
Bazel supports globally configuring optimization levels, e.g. via the `--compilation-mode` flag, so we should always be compiling our C dependencies with optimizations enabled for production builds.

Setting this for `lz4` specifically means we will _always_ compile with optimizations, even in debug builds. The compile time with optimizations enabled is not much longer than without, but the runtime perf is much higher, so this tradeoff is worth it.

Also I filed https://github.com/MaterializeInc/materialize/issues/27033 to make sure all of this is configured correctly